source/PreferencesPanel.cpp

```diff
@@ -178,6 +198,29 @@ bool PreferencesPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comma
 		if(zones[latest].Value().KeyName() != Command::MENU.KeyName())
 			Command::SetKey(zones[latest].Value(), 0);
 	}
+	else if(key == 'i' && page == 'p')
+	{
+		page = 'i';
+		if(!downloadedInfo)
+		{
+			PluginHelper::Download(PLUGIN_LIST_URL.c_str(), (Files::Config() + "plugins.json").c_str());
+			ifstream pluginlistFile(Files::Config() + "plugins.json");
+			pluginInstallList = nlohmann::json::parse(pluginlistFile);
+			pluginlistFile.close();
+			pluginInstallPages = ((pluginInstallList.size() - (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE))
+				/ MAX_PLUGIN_INSTALLS_PER_PAGE) + (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE > 0);
+			downloadedInfo = true;
+		}
+	}
+	else if(key == 'i' && page == 'i' && selectedPluginInstall.url.size())
+		installFeedbacks.emplace_back(Plugins::Install(selectedPluginInstall));
+	else if(key == 'u' && page == 'i' && selectedPluginInstall.url.size())
```

Comment:
Updating Plugins should be prohibited if the plugin is already at the latest version

Reply:
you sure?
this could be a way to repair a broken plugin if somebody messed something up

Reply:
If a plugin is broken I would delete and reinstall it. Using update to repair it seems weird to me idk

Reply:
done